Jenkinsfile

```diff
@@ -0,0 +1,74 @@
+#! groovy
+@Library('corda-shared-build-pipeline-steps@5.0.1') _
+
+import com.r3.build.enums.BuildEnvironment
+import com.r3.build.utils.GitUtils
+
+int cpus = 1
+BuildEnvironment buildEnvironment = BuildEnvironment.AMD64_LINUX
+
+GitUtils gitUtils = new GitUtils(this)
+
+pipeline {
+    agent {
+        kubernetes {
+            cloud "eks-e2e"
+            yaml kubernetesBuildAgentYaml('build', buildEnvironment, cpus)
+            idleMinutes 15
+            podRetention always()
+            nodeSelector ([
+                    "kubernetes.io/arch=${buildEnvironment.buildArchitecture.toString().toLowerCase()}",
+                    "kubernetes.io/os=${buildEnvironment.buildOperatingSystem.toString().toLowerCase()}",
+                    "role=jenkins-agent"
+            ].join(','))
+            label ([
+                    "kubernetes-build-agent",
+                    "${cpus}cpus",
+                    "${buildEnvironment.buildArchitecture.toString().toLowerCase()}",
+                    "${buildEnvironment.buildOperatingSystem.toString().toLowerCase()}"
+            ].join('-'))
+            showRawYaml false
+            defaultContainer 'build'
+        }
+    }
+
+  parameters {
+    booleanParam defaultValue: (gitUtils.isReleaseBranch() || gitUtils.isReleaseTag(env.TAG_NAME)), description: 'Publish artifacts to Artifactory?', name: 'DO_PUBLISH'

```

Comment:
Should be gitUtils.isReleaseTag()

Passing env.TAG_NAME would make this return true in 100% of tag builds, as i you look at that method implementation

```
    boolean isReleaseTag(String regexPattern = "(release/.*\$)|(^release_.*\$)|(^release-.*\$)") {
        return steps.env.TAG_NAME ? steps.env.TAG_NAME.matches(regexPattern) : false
    }
```

we are specifying the common release tag patters to match passing env.TAGNAME would mean any tag even TEST-TAG or something like this would trigger this to be true which is too broad, just go with the default parameter i.e leave blank and the regexpattern specified there will be used